mailet/standard/src/main/java/org/apache/james/transport/mailets/ContentReplacer.java

```diff
@@ -0,0 +1,130 @@
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.mailets;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.List;
+import java.util.regex.Matcher;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.ContentType;
+import javax.mail.internet.ParseException;
+
+import org.apache.mailet.Mail;
+import org.apache.mailet.MailetException;
+import org.apache.mailet.base.GenericMailet;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Strings;
+
+public class ContentReplacer {
+
+    private final boolean debug;
+    private final GenericMailet logOwner;
+
+    public ContentReplacer(boolean debug, GenericMailet logOwner) {
+        this.debug = debug;
+        this.logOwner = logOwner;
+    }
+
+    public String applyPatterns(List<ReplacingPattern> patterns, String text) {
+        for (ReplacingPattern replacingPattern : patterns) {
+            boolean changed;
+            do {
+                changed = false;
+                Matcher matcher = replacingPattern.getMatcher().matcher(text);
+                if (matcher.find()) {
+                    String replaced = matcher.replaceAll(replacingPattern.getSubstitution());
+                    if (debug) {
+                        logOwner.log("Subject rule match: " + replacingPattern.getMatcher());
+                    }
+                    text = replaced;
+                    changed = true;
+                }
+            } while (replacingPattern.isRepeat() && changed);
+        }
+        
+        return text;
+    }
+
+    public void applyPattern(Mail mail, ReplaceConfig replaceConfig, Optional<Charset> charset) throws MailetException {
+        try {
+            boolean subjectChanged = applySubjectReplacingUnits(mail, replaceConfig, charset);
+            boolean contentChanged = applyBodyReplacingUnits(mail, replaceConfig, charset);
+
+            if (charset.isPresent() && !contentChanged) {
```

Comment:
I think it's very useful but using ReplaceContent for forcing the encoding looks like a hack.
We can change the document (and the behavior) IMO